draft-ietf-rtcweb-jsep.xml

```diff
@@ -576,6 +576,135 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
+            types are used, and there is one "a=imageattr" line per payload
+            type.</t>
+
+            <t>The "a=imageattr" field can contain multiple sets of parameters,
+            each with a specified q value (a preference). In this specification,
+            the preferences are used in the following manner:</t>
+
+            <t><list style="symbols">
+              <t>q=0.1 is used to indicate the decode capabilities of the
+              device, if known. Using image sizes outside these ranges
+              risks decode failure.</t>
+
+              <t>q=0.5 is used to indicate the limits to image resolution
+              caused by mandatory constraints set by the user (if any).</t>
+
+              <t>q=1.0 is used to indicate the preferred format. Usually the
+              present format of the videostream will be preferred by the
+              sender and the size of the receiver's display surface will be
+              preferred by the receiver; if this format can be used, the video
+              stream can be passed through with an absolute minimum of
+              rescaling. If the user has specified "ideal" constraint values,
+              these might be used instead.</t>
+            </list>
+            Thus, a system with a HD decoder might send</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]</t>
+              </list>If the Javascript has constrained the stream to have
+            width={min:320, max:640}, the result might be</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]
+              [x=[320:640],y=[16:1080],q=0.5]</t>
+              </list>and if the stream is known to be displayed on a surface of
+            dimension 640x480, the result might be</t>
+
+            <t><list style="symbols">
+              <t>a=imageattr:* recv [x=[16:1920],y=[16:1080],q=0.1]
+              [x=[320:640],y=[16:1080],q=0.5] [x=640,y=480,q=1.0]</t>
+            </list>
+            The "par=" parameter is used to represent a constraint on
+            the aspect ratio. The numbers given are produced from the
+            user-specified value by rounding the minimum down to four decimal
+            digits, and rounding the maximum up to four decimal digits.</t>
+          </section>
+
+          <section title="Choice of imageattr send attributes">
+            <t>There are two cases for generation of the send
+            attributes: When the correspondent's recv attribute is
+            unknown, and when the recv attribute is known.
+            </t>
+            <t>When the recv attribute is unknown, such as for an
+            initial offer, the send attribute is generated as follows:
+            </t>
+            <t><list style="symbols">
+              <t>q=0.1 is used to indicate the range of video sizes the
```

Comment:
This may not be known well in the case of a software encoder, so we may just pick some arbitrary max (e.g. 4K)